app/src/main/java/com/spendster/presentation/addExpenses/chooseCategory/ChooseCategoryPresenter.java

```diff
@@ -0,0 +1,19 @@
+package com.spendster.presentation.addExpenses.chooseCategory;
+
+public class ChooseCategoryPresenter {
+    private final CategoryView categoryView;
+    private final CategoryModel categoryModel;
+
+    public ChooseCategoryPresenter(CategoryView categoryView, CategoryModel categoryModel) {
+        this.categoryView = categoryView;
+        this.categoryModel = categoryModel;
+    }
+
+    public void fetchCategories() {
+        if (categoryModel.storeCategories() != null) {
```

Comment:
@SimonMaha categories should be extracted into an object here. As now, you are calling gen method twice.